internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,429 @@
+	return &atlas.ProviderSettings{
+		InstanceSizeName: tier,
+		ProviderName:     opts.Provider,
+		RegionName:       opts.Region,
+	}
+}
+
+// askDBUserAccessListFlags allows the user to set required flags by using interactive prompts
+func (opts *Opts) askDBUserAccessListFlags() error {
+	qs := opts.newDBUserQuestions()
+
+	if q := opts.newAccessListQuestion(); q != nil {
+		qs = append(qs, q)
+	}
+
+	if len(qs) > 0 {
+		if err := survey.Ask(qs, opts); err != nil {
+			return err
+		}
+	}
+
+	if opts.DBUserPassword == "" {
+		pwd, err := randgen.GenerateRandomBase64String(passwordLength)
+		if err != nil {
+			return err
+		}
+		opts.DBUserPassword = pwd
+	}
+
+	return nil
+}
+
+func (opts *Opts) askClusterFlags() error {
+	qs := opts.newClusterQuestions()
+
+	if err := survey.Ask(qs, opts); err != nil {
+		return err
+	}
+
+	if regionQ := opts.newRegionQuestions(); regionQ != nil {
+		// we call survey.Ask two times because the region question needs opts.Provider to be populated
+		if err := survey.Ask([]*survey.Question{regionQ}, opts); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (opts *Opts) newDBUserQuestions() []*survey.Question {
+	var qs []*survey.Question
+	if opts.DBUsername == "" {
+		message := "Insert the Username for authenticating to MongoDB"
+		usrDefault := dbUsername()
+		if usrDefault != "" {
+			message = fmt.Sprintf("Insert the Username for authenticating to MongoDB [Press Enter to use '%s']", usrDefault)
+		}
+		q := &survey.Question{
+			Name: "dbUsername",
+			Prompt: &survey.Input{
+				Message: message,
+				Help:    usage.DBUsername,
+				Default: usrDefault,
+			},
+			Validate: func(val interface{}) error {
+				username, _ := val.(string)
+				user, err := opts.store.DatabaseUser(convert.AdminDB, opts.ConfigProjectID(), username)
+				if err != nil {
+					if !strings.Contains(err.Error(), fmt.Sprintf("No user with username %s exists.", username)) {
+						return err
+					}
+				}
+
+				if user != nil {
+					return errors.New("a user with this username already exists")
+				}
+
+				return nil
+			},
+		}
+
+		qs = append(qs, q)
+	}
+
+	if opts.DBUserPassword == "" {
+		q := &survey.Question{
+			Name: "dbUserPassword",
+			Prompt: &survey.Password{
+				Message: "Insert the Password for authenticating to MongoDB [Press Enter to use an auto-generated password]",
+				Help:    usage.Password,
+			},
+		}
+
+		qs = append(qs, q)
+	}
+
+	return qs
+}
+
+func (opts *Opts) newAccessListQuestion() *survey.Question {
+	if opts.IPAddress != "" {
+		return nil
+	}
+
+	publicIP := net.IPAddress()
+	message := "Insert the IP entry to add to the Access List"
+	if publicIP != "" {
+		message = fmt.Sprintf("Insert the IP entry to add to the Access List [Press Enter to use your public IP address '%s']", publicIP)
+	}
+
+	return &survey.Question{
+		Name: "ipAddress",
+		Prompt: &survey.Input{
+			Message: message,
+			Help:    usage.AccessListIPEntry,
+			Default: publicIP,
```

Comment:
I am not sure why lint is failing given that `publicIP` is used also outside the if

Reply:
I don't get this error when running `make lint`

Reply:
your version of the linter is older probably also this may be a false possitive

Reply:
I have added` //nolint`